core/src/price_estimation/kraken.rs

```diff
@@ -49,6 +50,16 @@ where
     /// that are used when computing the price map.
     #[allow(clippy::needless_lifetimes)]
     async fn get_token_asset_pairs(&self, tokens: &[TokenId]) -> Result<HashMap<String, TokenId>> {
+        let token_infos: Vec<_> = stream::iter(tokens)
```

Comment:
I feel like this snippet it repeated across `PriceSource` implementations. Does it make sense for just `PriceOracle` to fetch all the on-chain token data and then just pass it as a parameter to each `PriceSource`?

Reply:
It depends on which tokens are listed on the exchange (e.g. if we have 100 💩 coins we might only fetch token data for a the small subset that is listed on kraken & dex.AG)

Reply:
Makes sense. I think if we add a caching layer as you suggested it will be fine.